### PR TITLE
add do get role api

### PR DESCRIPTION
**Pull request by runzexia**
add api to get role's permission & pattern

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
will try to review and release it this week

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java [hunk 1]
```diff
@@ -310,6 +310,41 @@ public void doAddRole(@QueryParameter(required = true) String type,
...
+        if (roleMap != null){
+            Role role = roleMap.getRole(roleName);
+            if (role != null){
+                Set<Permission> permissions = role.getPermissions();
+                Map<String,Boolean> permissionsMap = new HashMap<String, Boolean>();
+                for (Permission permission : permissions) {
+                    permissionsMap.put(permission.getId(),permission.getEnabled());
+                }
+                responseJson.put("permissionIds",permissionsMap);
+                if (type.equals(RoleBasedAuthorizationStrategy.PROJECT)){
```
Comment (oleg-nenashev) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java]: It also applies to agents and other types. I would rather check for `!= RoleBasedAuthorizationStrategy.GLOBAL`

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java [hunk 2]
```diff
@@ -310,6 +310,41 @@ public void doAddRole(@QueryParameter(required = true) String type,
...
     }
 
+    /**
+     * API method to get role.
+     * Example: {@code curl -XGET 'http://localhost:8080/jenkins/role-strategy/strategy/getRole
+     * ?type=globalRoles&roleName=admin'}
+     *
+     * @param type (globalRoles, projectRoles, slaveRoles)
+     * @param roleName name of role (single, no list)
+     * @throws IOException
```
Comment (oleg-nenashev) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java]: Better to add some documentation so there is no Javadoc warnings

**Comment by runzexia** [comment]
fix it @oleg-nenashev

**Comment by runzexia** [comment]
@oleg-nenashev Could you please help review this again? Any comments is appreciated. Thanks!

**Comment by oleg-nenashev** [comment]
I will release the new version soon, do not want to make it wait for long

**Comment by runzexia** [comment]
Thanks! @oleg-nenashev

**Pull request merged**